src/storage/src/hummock/compactor/mod.rs

```diff
@@ -225,6 +227,78 @@ impl Compactor {
             .await;
         let multi_filter_key_extractor = Arc::new(multi_filter_key_extractor);
 
+        let sstable_ids = compact_task
+            .input_ssts
+            .iter()
+            .flat_map(|level| level.table_infos.iter())
+            .map(|table_info| table_info.id)
+            .collect_vec();
+
+        let compaction_size = compact_task
+            .input_ssts
+            .iter()
+            .flat_map(|level| level.table_infos.iter())
+            .map(|table_info| table_info.file_size)
+            .sum::<u64>();
+
+        let sstable_size = (context.options.sstable_size_mb as u64) << 20;
+        if compaction_size > sstable_size {
```

Comment:
when `compaction_size < sstable_size`, the splits of compact_task will not be assigned, its it expected ? It will panic when splits is empty

Reply:
It shall be a default value.
vec![KeyRange::inf()]

Reply:
We must remove code for `splits` in `src/meta/` and leave it with a default value

Reply:
we can remove the max_sub_compaction config after we introduced the `generate_splits` in `compactor`